src/Microsoft.Framework.PackageManager/Restore/RestoreOperations.cs

```diff
@@ -314,7 +324,26 @@ public Task<GraphItem> FindLibraryCached(RestoreContext context, LibraryRange li
             }
 
             WalkProviderMatch bestMatch = null;
-            var matches = await Task.WhenAll(tasks);
+            var matches = new List<WalkProviderMatch>();
+
+            // Short circuit if we find an exact match
+            while (tasks.Any())
+            {
+                var task = await Task.WhenAny(tasks);
```

Comment:
Ahh this was the change you were speaking of. Could you also send a PR to NuGet with this change?